test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenTransactSpecs.java

```diff
@@ -430,6 +445,215 @@ public HapiApiSpec balancesChangeOnTokenTransfer() {
+						getReceipt("mintTransferTxn").logged()
+				);
+	}
+
+	public HapiApiSpec uniqueTokenTxnWithNoAssociation() {
+		return defaultHapiSpec("UniqueTokenTxnWithNoAssociation")
+				.given(
+						cryptoCreate(TOKEN_TREASURY).balance(0L),
+						cryptoCreate(FIRST_USER).balance(0L),
+						newKeyNamed("supplyKey"),
+						tokenCreate(A_TOKEN)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.supplyType(TokenSupplyType.INFINITE)
+								.initialSupply(0)
+								.supplyKey("supplyKey")
+								.treasury(TOKEN_TREASURY)
+				)
+				.when(
+						mintToken(A_TOKEN, List.of(ByteString.copyFromUtf8("memo")))
+				)
+				.then(
+						cryptoTransfer(
+								movingUnique(1, A_TOKEN).between(TOKEN_TREASURY, FIRST_USER)
+
+						).hasKnownStatus(FAIL_INVALID)
+				);
+	}
+
+	public HapiApiSpec uniqueTokenTxnWithFrozenAccount() {
+		return defaultHapiSpec("UniqueTokenTxnWithFrozenAccount")
+				.given(
+						cryptoCreate(TOKEN_TREASURY).balance(0L),
+						cryptoCreate(FIRST_USER).balance(0L),
+						newKeyNamed("freezeKey"),
+						tokenCreate(A_TOKEN)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.supplyType(TokenSupplyType.INFINITE)
+								.initialSupply(0)
+								.freezeKey("freezeKey")
+								.freezeDefault(true)
+								.supplyKey("freezeKey")
+								.treasury(TOKEN_TREASURY),
+						tokenAssociate(FIRST_USER, A_TOKEN)
+				)
+				.when(
+						mintToken(A_TOKEN, List.of(ByteString.copyFromUtf8("memo")))
+				)
+				.then(
+						cryptoTransfer(
+								movingUnique(1, A_TOKEN).between(TOKEN_TREASURY, FIRST_USER)
+
+						)
+								.hasKnownStatus(ACCOUNT_FROZEN_FOR_TOKEN)
+				);
+	}
+
+	public HapiApiSpec uniqueTokenTxnWithSenderNotSigned() {
+		return defaultHapiSpec("UniqueTokenTxnWithOwnerNotSigned")
+				.given(
+						newKeyNamed("supplyKey"),
+						newKeyNamed("signingKeyTreasury"),
+						newKeyNamed("signingKeyFirstUser"),
+						cryptoCreate(TOKEN_TREASURY).key("signingKeyTreasury"),
+						cryptoCreate(FIRST_USER).key("signingKeyFirstUser"),
+						tokenCreate(A_TOKEN)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.supplyType(TokenSupplyType.INFINITE)
+								.initialSupply(0)
+								.supplyKey("supplyKey")
+								.treasury(TOKEN_TREASURY),
+						tokenAssociate(FIRST_USER, A_TOKEN)
+				)
+				.when(
+						mintToken(A_TOKEN, List.of(ByteString.copyFromUtf8("memo")))
+				)
+				.then(
+						cryptoTransfer(
+								movingUnique(1, A_TOKEN).between(TOKEN_TREASURY, FIRST_USER)
+						)
+								.signedBy(DEFAULT_PAYER)
+								.hasKnownStatus(INVALID_SIGNATURE)
+				);
+	}
+
+	public HapiApiSpec uniqueTokenTxnWithReceiverNotSigned() {
+		return defaultHapiSpec("UniqueTokenTxnWithOwnerNotSigned")
+				.given(
+						newKeyNamed("supplyKey"),
+						newKeyNamed("signingKeyTreasury"),
+						newKeyNamed("signingKeyFirstUser"),
+						cryptoCreate(TOKEN_TREASURY).key("signingKeyTreasury"),
+						cryptoCreate(FIRST_USER).key("signingKeyFirstUser"),
```

Comment:
This account must have `receiver signature required` property set to `true`, then when we do `cryptoTransfer` TO that account signed only by the `payer` and `treasury`, it must fail with `INVALID_SIGNATURE`

Reply:
It is corrected!